src/analysis/souffle/examples/vision_pipeline/VisionPipelineTagOwners.dl

```diff
@@ -13,19 +13,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //-----------------------------------------------------------------------------
-#include "authorization_logic.dl"
+#include "../../authorization_logic.dl"
 
 // This code assign owners to tags for the VisionPipeline.arcs example.
-// This is not generated code. This is policy code that is written directly
-// in datalog (for now).
-//
-// This is the only bit of "policy" code that
-// isn't expressed in authorization logic, and it isn't in authorization logic
-// just because these facts are "universally true" rather than having a 
-// "speaker". We might also try making these "SomePrincipal says ownsTag(...)"
-// instead later, but this seems like a simpler way to start.
-ownsTag("EndUser", "raw_video_tag").
-ownsTag("EndUser", "detected_images_tag").
-ownsTag("EndUser", "user_selection_tag").
-ownsTag("EndUser", "product_id_tag").
-ownsTag("ServiceProvider", "image_detection_model_tag").
+// It comes from authorization logic where it looks like "P says createTag(t)"
+saysCreateTag("EndUser", "raw_video_tag").
```

Comment:
(Context: this PR is just a "hack" to allow tag ownership, application of tags to be produced by authorization logic and to have a speaker. Because it's just a hack, it's not really a proper design).

All the facts from authorization logic start with `says_`, so to have tag ownership come from authorization logic, this needs to involve a relation that starts with says. Before this PR "ownsTag" was not a fact with a speaker, it was just globally believed by everyone.

Another way to do this could be

```
.decl says_OwnsTag(p: Principal, t: Tag)
ownsTag(principal, tag) :- saysOwnsTag(principal, principal, t).
```

This does not stop an adversarial principal from taking ownership of a tag owned by another one, though. A way we can at least make this similar to the case where tag ownership is a "unversal truth" is to have this rule and some other rule that says the whole policy fails if there are multiple principals that own the same tag. Roughly:

```
FAIL() :- ownsTag(p1, tag), ownsTag(p2, tag), p1 != p2
```

(There are probably even more aspects of giving speakers to tag ownership / applying tags to handles that would need to be fleshed out more if we want to keep this.)

Reply:
I made this `saysOwnsTag(speaker, owner, tag)`, just so that it is possible to pull this in from the auth logic code for now. Hopefully the name `saysOwnsTag` makes its relationship with `ownsTag` more clear.

Fine to leave it for now?

Reply:
Marking this resolved but do feel free to unresolve (I think that's our procedure, right?) :)